src/Page.js

```diff
@@ -482,7 +488,7 @@ Page.prototype.insertSiteNav = function (pageData) {
     throw new Error(`More than one <navigation> tag found in ${siteNavPath}`);
   } else if (siteNavDataSelector('navigation').length === 1) {
     const siteNavHtml = md.render(siteNavDataSelector('navigation').html().trim().replace(/\n\s*\n/g, '\n'));
-    const formattedSiteNav = formatSiteNav(siteNavHtml);
+    const formattedSiteNav = formatSiteNav(siteNavHtml, this.src);
```

Comment:
The first time we render the siteNav for a page, we pass in `this.src` so that `formatSiteNav` can locate the link to set `class=current` to at `Page.js#143`. When we recursively format the same siteNav for nested lists in `Page.js#157` and `Page.js#184`, we do not pass in the `src` parameter since we do not need to highlight the same link repeatedly for the same page.